### DISCUSSION TITLE
Ark  ActiveEvent changer

### DISCUSSION THREAD

**Discussion by Nytridr** [gameserverapp/Platform#421] [Suggestions]
I run a script on one of my other servers to change the active event on a certain day.

Was wondering if there is a way to do this with GSA also.
Its pretty simple.. Just change it in the config file.  Not sure where GSA keeps the configs we save on the dashboard so not sure if this would work on the server side or not.

`
#!/bin/bash

DATETIME=`date "+%Y.%m%d.%H%M.%S"`
LOGFILEERROR=/home/steam/Servers/Logs/ActiveEventChanger-ERROR-${DATETIME}.log
LOGFILE=/home/steam/Servers/Logs/ActiveEventChanger-${DATETIME}.log

oldEvent='none'
NewEvent='none'
source /etc/arkmanager/arkmanager.cfg
echo $arkopt_ActiveEvent
echo "$oldEvent  ---  $NewEvent"
	if [ $arkopt_ActiveEvent == 'Easter' ]
	then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='Arkaeology'
        elif [ $arkopt_ActiveEvent == 'Arkaeology' ]
		then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='ExtinctionChronicles'
        elif [ $arkopt_ActiveEvent == 'ExtinctionChronicles' ]
		then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='WinterWonderland'
        elif [ $arkopt_ActiveEvent == 'WinterWonderland' ]
		then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='vday'
        elif [ $arkopt_ActiveEvent == 'vday' ]
		then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='Summer'
        elif [ $arkopt_ActiveEvent == 'Summer' ]
		then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='FearEvolved'
        elif [ $arkopt_ActiveEvent == 'FearEvolved' ]
		then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='TurkeyTrial'
        elif [ $arkopt_ActiveEvent == 'TurkeyTrial' ]
		then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='birthday'
        elif [ $arkopt_ActiveEvent == 'birthday' ]
		then
                oldEvent=$arkopt_ActiveEvent
                NewEvent='Easter'
        fi
echo "${DATETIME} - $oldEvent changed to $NewEvent"
echo "${DATETIME} - Changing from ${oldEvent} to ${NewEvent}" 2>>${LOGFILEERROR}

sed -i.BKUP "s/arkopt_ActiveEvent=\"$oldEvent"/arkopt_ActiveEvent=\"$NewEvent"/g" /etc/arkmanager/arkmanager.cfg

/usr/local/bin/arkmanager restart  --saveworld --warn --warnreason='Event has been changed from ${oldEvent} to ${NewEvent}.  Events will change every Tuesday night at midnight server time.' @all

`

**Comment by mvaessen** [comment]
Hi,

This is the suggestion section, but I think your post is mostly a question. Please do not use the suggestion board for questions and please do not `@all`. This can get you banned from our Github.

GSA does not have similar settings like the software you're mentioning. It's a completely different system.
At the moment there are no plans to automatically set active events for ARK. You can program this using  Tasks if you like to automate this.

**Reply by Nytridr** [reply]
actually this is ran through crontab as a bash script.  and I have looked through the tasks on GSA but I could not find a way to do it, which is why I put this in suggestions, but I also added the question to it just in-case there was a way, and I just did not see it.  The only thing I seen was changing config templates which would mean 9 different config templates if I am understanding things correctly which I would say is very inconvenient if that is the way it is done.

Simple fix would be to have event list added to the task and then would be auto added at run time.

I would just change it in the files locally, but I am not sure how GSA stores its stuff as in.   are the configs stored on the server itself? or does GSA run the servers and use what is saved on GSA servers, which would mean there is no saved configs until run time.

oh and the AT all was in code and was part of the script as what is being used and was listed as code also.

**Reply by mvaessen** [reply]
Im hoping to start working on #282 in a couple weeks (no specific ETA yet) which will include options to setup variables in templates.
At the moment is GSA not tracking what events are active for ARK. This is something we think admins can do perfectly fine.

Making changes to the configs locally wont work because GSA automatically replaces your configs on each start/restart to ensure your game server never loses a config setting. The best option at the moment is to simply add the `-activeEvent=X` when an event is active.

Since the ARK developers announced EOL for the first ARK version (non-facelift) would it surprise me if the events will continue after that happens.

**Reply by Nytridr** [reply]
that would be awesome if this made it into it.  And thank you for explaining things.

**Reply by birdman3131** [reply]
It is possible beacon could handle this but ive not messed with it enough to try.
